src/zenml/zen_server/zen_server_api.py

```diff
@@ -265,6 +298,12 @@ async def dashboard(request: Request) -> Any:
     return templates.TemplateResponse("index.html", {"request": request})
 
 
+@app.get("/system/last-activity")
+async def get_last_activity() -> Dict[str, str]:
+    """Get the timestamp of the last user activity."""
+    return {"last_user_activity": last_user_activity.isoformat()}
+
+
```

Comment:
I'll leave some corner-cases for you to consider here:

* when you have a server running multiple pod replicas, this will be less reliable, because calling the endpoint will only return the time when one particular pod has been accessed. But with larger time periods this should even out, so a few seconds or even hours of time difference between pods won't matter.
* the timestamp is reset whenever the pod is reset. This basically means that if even one of the zenml server pods is restarted during an observation time window, it will incorrectly appear as if it has seen user activity. Pods can be re-started purely for maintenance reasons, like pods being moved to another Kubernetes node to free up resources.

Reply:
Thanks! I thought about both cases and I think that for our use-case this should be fine

/re replicas, it should even out, we're  going for a 12 hour horizon anyway
/re restarts, yeah it will just pick up the latest timestamp. this shouldn't effect a lot of things in the grand scheme of things.